### PR TITLE
#173766649 edit user endpoint

### PR DESCRIPTION
**Pull request by tobieeniafe**
#### What does this PR do?
Add edit user endpoint for dashboard

#### Description of Task to be completed?
Create endpoint to edit a user

#### How should this be manually tested?
`PUT /api/v1/user/update`
Sample data
`{
  "firstName": "User Firstname",
  "lastName": "User Lastname",
  "phone": "08012345678"
}`

#### Any background context you want to provide?
All requests should be sent with token as header `Authorization`

#### What are the relevant pivotal tracker stories?
#173766649

### CODE REVIEW

File: server/controllers/user.controllers.js [hunk 1]
```diff
@@ -83,6 +84,16 @@ const UserController = {
       .status(httpStatus.OK)
       .json({ success: true, message: 'Your information has been successfully retrieved', user });
   },
+
+  update(req, res, next) {
+    const updateduser = req.locals;
+    const { user } = req;
+    updateUser({ ...updateduser, id: user._id })
+      .then(() => {
+        res.status(httpStatus.OK).json({ success: true, message: 'User updated' });
```
Comment (andela-hpopoola) [inline server/controllers/user.controllers.js:93]: Can you return the updated user information here?

File: test/controllers/user.controller.test.js [hunk 2]
```diff
@@ -601,3 +601,75 @@ describe('Current User', () => {
...
+    it('returns a updated user', (done) => {
+      request()
+        .put('/api/v1/user/update')
+        .set('authorization', token)
+        .send(newUser)
+        .end((err, res) => {
+          expect(res).to.have.status(200);
+          expect(res.body.success).to.be.eql(true);
+          expect(res.body).to.have.property('updateduser');
+          done();
```
Comment (andela-hpopoola) [inline test/controllers/user.controller.test.js:629]: You need to add tests to verify the firstName, lastName and phone number has truly been updated.

File: test/controllers/user.controller.test.js [hunk 3]
```diff
@@ -601,3 +601,75 @@ describe('Current User', () => {
...
+
+  context('without token', () => {
+    it('returns error', (done) => {
+      request()
+        .put('/api/v1/user/update')
+        .send(newUser)
+        .end((err, res) => {
+          expect(res).to.have.status(403);
+          expect(res.body.success).to.be.eql(false);
+          done();
```
Comment (andela-hpopoola) [inline test/controllers/user.controller.test.js:642]: Can add test for the message or error that is returned to be sure that the right message is returned.

File: test/controllers/user.controller.test.js (hunk 2 above)
Comment (tobieeniafe) [reply test/controllers/user.controller.test.js:629]: the scenario to making sure the values were updated was tested for in the `user.service.test.js`. Line `380`. If I understand you correctly.

**Pull request merged**